presentation/src/main/java/com/example/presentation/holder/SearchViewHolder.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.presentation.holder
+
+import androidx.recyclerview.widget.RecyclerView
+import com.example.presentation.databinding.ItemSerachUserBinding
+import com.example.presentation.setImageUrl
+import model.SearchModel
+
+class SearchViewHolder(
+    private val binding: ItemSerachUserBinding,
+    private val itemClickListener: (SearchModel) -> Unit,
+) : RecyclerView.ViewHolder(binding.root) {
+
+    fun bind(item: SearchModel) {
+        binding.apply {
+            root.setOnClickListener {
+                itemClickListener(item)
+            }
```

Comment:
init 블록에서 달고 싶은데 viewBinding이 오랜만이라 init에서 item을 가져오는게 어렵네요. 혹시 방법 있으면 공유점요 ㅎㅎ

Reply:
이 블로그 참고해보시면 좋을 것 같습니당
https://mangbaam.github.io/android/2022/10/02/android-recyclerview-click.html#h-setonclicklistener-%ED%98%B8%EC%B6%9C-%ED%9A%9F%EC%88%98-%EB%B9%84%EA%B5%90

Reply:
아쉽게도.. 해당블로그는 단순 포지션(int)값을 가지고 리스너를 달고 있네용ㅠㅠ 저는 뷰홀더에 들어가는 item자체를 가져오고 싶은겁니당

Reply:
기존 dataBinding에서는 layout variable 에 선언된 아이템을 가져왔는데 viewBinding은 모르겠네요

Reply:
```suggestion
            root.setOnClickListener {
                itemClickListener(currentList[bindingAdapterPosition])
            }
```
위 방식은 어떨까요? 
ListAdapter#getCurrentList를 통해 List를 받아올 수 있고, position을 통해 아이템 객체를 받아올 수 있습니다. 다만, ViewHolder에서 쓰기 위해서는 생성자로 `currentList`를 전달할 필요가 있겠네용 ㅎㅎ

Reply:
init {
            val item = getItem(adapterPosition)
     }

이렇게 가져오면 오류날까용,,,?! 저도 해보진 않았는데 쉽게 가져올 수 있을 것 같긴해서!

Reply:
getItem()이 없네요ㅠㅠ

Reply:
@ddyeon `adapterPosition`은 Deprecated되었다고 하네용 ;;ㅠㅠ

Reply:
그러게여.. dataBinding 말고 viewBinding은 잘 모르겠네..
이런건 어때여?

```kotlin
private var item: SearchModel? = null

init {
    binding.root.setOnClickListener {
        itemClickListener(item ?: return@setOnClickListener)
    }
}

fun bind(item: SearchModel) {
    this.item = item
    ...
}
```